src/components/ResultList/Item/Item.tsx

```diff
@@ -66,6 +72,111 @@ export const Item = (props: IItemProps): ReactElement => {
     </Link>
   ) : null;
 
+  // full text sources and data
+
+  let fullSourceItems: ItemType[] = [];
+
+  let dataProductItems: ItemType[] = [];
+
+  if (!hideActions && doc.esources) {
+    const sources = processLinkData(doc, null);
+
+    const fullTextSources = sources.fullTextSources;
+
+    const dataProducts = sources.dataProducts;
+
+    fullSourceItems = fullTextSources.map((source) => ({
+      id: source.name,
+      text: source.name,
+      label: source.name,
+      path: source.url,
+      domId: `fullText-${source.name}`,
+      newTab: true,
+    }));
+
+    dataProductItems = dataProducts.map((dp) => ({
+      id: dp.name,
+      text: dp.name,
+      label: dp.name,
+      path: dp.url,
+      domId: `dataProd-${dp.name}`,
+      newTab: true,
+    }));
+  }
+
+  const fullTextSourcesLabel = (
+    <>
+      {fullSourceItems.length > 0 ? (
+        <DocumentTextIcon
+          className="default-icon default-link-color cursor-pointer"
+          aria-label="Full text sources"
+          role="list"
+        />
+      ) : (
+        <DocumentTextIcon className="default-icon text-gray-300" aria-label="No Full text sources" role="list" />
+      )}
+    </>
+  );
+
+  const dataProductLabel = (
+    <>
+      {dataProductItems.length > 0 ? (
+        <DatabaseIcon
+          className="default-icon default-link-color cursor-pointer"
+          aria-label="Data products"
+          role="list"
+        />
+      ) : (
+        <DatabaseIcon className="default-icon text-gray-300" aria-label="No data products" role="list" />
+      )}
+    </>
+  );
+
+  // citations and references
+
+  const num_references =
+    !hideActions && doc['[citations]'] && typeof doc['[citations]'].num_references === 'number'
+      ? doc['[citations]'].num_references
+      : 0;
+
+  const num_citations =
+    !hideActions && doc['[citations]'] && typeof doc['[citations]'].num_citations === 'number'
+      ? doc['[citations]'].num_citations
+      : 0;
+
+  const referenceItems: ItemType[] = [];
+  if (num_citations > 0) {
+    referenceItems.push({
+      id: 'citations',
+      domId: `ref-dropdown-cit-${doc.bibcode}`,
+      label: `Citations (${num_citations})`,
+      path: `/abs/${bibcode}/citations`,
+    });
+  }
+
+  if (num_references > 0) {
+    referenceItems.push({
+      id: 'references',
+      domId: `ref-dropdown-ref-${doc.bibcode}`,
+      label: `References (${num_references})`,
+      path: `/abs/${bibcode}/references`,
+    });
+  }
+
+  const referencesLabel = (
+    <>
+      {referenceItems.length > 0 ? (
+        <ViewListIcon
+          className="default-icon default-link-color cursor-pointer"
+          aria-label="References and citations"
+          role="list"
+        />
+      ) : (
+        <ViewListIcon className="default-icon text-gray-300" aria-label="No references and citations" role="list" />
+      )}
+    </>
+  );
+
```

Comment:
I think this unnecessarily complicates the `<Item/>` component, I think we can at least offload all this logic into a separate component or hook to clean this up a bit

Reply:
Moved the dropdowns into a separate component